app/services/forest_liana/oidc_configuration_retriever.rb

```diff
@@ -0,0 +1,39 @@
+module ForestLiana
+  class OidcConfigurationRetriever
+    DEFAULT_EXPIRATION_IN_SECONDS = 30 * 60
+
+    def self._fetch_configuration()
+      response = ForestLiana::ForestApiRequester.get('/oidc/.well-known/openid-configuration')
+      if response.is_a?(Net::HTTPOK)
+        return JSON.parse(response.body)
+      else
+        raise ForestLiana::Errors::HTTP500Error.new(ForestLiana::MESSAGES[:SERVER_TRANSACTION][:OIDC_CONFIGURATION_RETRIEVAL_FAILED])        
+      end
+    end
+
+    def self.retrieve()
+      if @cached_well_known_configuration &&
+         @cached_well_known_configuration[:expiration] < Time.now.to_i 
+        clear_cache()
+      end
+  
+      if !@cached_well_known_configuration
+        begin
+          configuration = _fetch_configuration()
+          expiration_duration = DEFAULT_EXPIRATION_IN_SECONDS
+          expiration = Time.now + expiration_duration
+          @cached_well_known_configuration = { :configuration => configuration, :expiration => expiration }
+          
+        rescue => error
+          @cached_well_known_configuration = {}
+          raise error
+        end
+      end
+      return @cached_well_known_configuration[:configuration];
+    end
+  
+    def self.clear_cache()
+      @cached_well_known_configuration = {}
```

Comment:
[MISC] As there is cache here, you might be unable to make the all class a static one. However, if this cache is common to the all liana, you could instantiate it at start and make the class static nonetheless. (I'm available if unclear)